tests/pd/mock/mocker/service.rs

```diff
@@ -146,17 +142,37 @@ impl PdMocker for Service {
         }
     }
 
-    fn get_region_by_id(&self, req: &GetRegionByIDRequest) -> Option<Result<GetRegionResponse>> {
+    fn get_region(&self, req: &GetRegionRequest) -> Option<Result<GetRegionResponse>> {
         let mut resp = GetRegionResponse::new();
-        let mut region = Region::new();
-        let region_path = make_region_key(req.region_id);
+        let key = req.get_region_key();
+        let regions = self.regions.lock().unwrap();
+
+        for region in regions.values() {
+            if key >= region.get_start_key() &&
```

Comment:
Does it compatible with rocksdb's key comparator?

Reply:
Yep, after all, we use the same way to compare keys everywhere.